File: algorithms/easy/remove_letter_to_equalize_frequency.cpp
bool isEqual(vector<int> v) {
  vector<int> erased;
  copy_if(begin(v), end(v), back_inserter(erased), [](int x) { return x != 0; });
  set<int> s(erased.begin(), erased.end());
  return s.size() == 1;
}

class Solution {
public:
  bool equalFrequency(string word) {
    vector<int> map(26, 0);
      
    for (auto c : word) {
      map[c-'a']++;
    }
    for (int i = 0; i < 26; i++) {
      if (map[i] == 0) continue;
      map[i]--;
      if (isEqual(map)) {
        return true;
      }
      map[i]++;
    }
    return false;
  }
};

File: algorithms/easy/remove_letter_to_equalize_frequency.rs
use std::collections::HashSet;

fn is_equal(arr: &[u8]) -> bool {
    let set: HashSet<_> = arr.iter().filter(|v| **v != 0).collect();
    set.len() == 1
}

impl Solution {
    pub fn equal_frequency(word: String) -> bool {
        let mut map = vec![0; 26];
        for b in word.as_bytes() {
            map[(b - b'a') as usize] += 1;
        }

        for i in 0..map.len() {
            if map[i] > 0 {
                map[i] -= 1;
                if is_equal(&map) {
                    return true;
                }
                map[i] += 1;
            }
        }

        false
    }
}
